### ISSUE TITLE
manually fileset deletion before pvc deletion operation does not cleanup symlink path

### ISSUE THREAD

**Issue by saurabhwani5** [IBM/ibm-spectrum-scale-csi#939]
## Describe the bug
After creating a PVC and if fileset is deleted & after that we delete the pvc then  symlink path of the fileset won't get deleted.

## How to Reproduce?

1. Install 2.9.0 CSI (dev)
```
[root@saurabh-csi-master 0203]# oc apply -f csiscaleoperators.csi.ibm.com_cr.yaml
csiscaleoperator.csi.ibm.com/ibm-spectrum-scale-csi created
[root@saurabh-csi-master 0203]# oc get pods
NAME                                                  READY   STATUS    RESTARTS   AGE
ibm-spectrum-scale-csi-2hq5h                          3/3     Running   0          39s
ibm-spectrum-scale-csi-attacher-5885c459cb-j9d7g      1/1     Running   0          39s
ibm-spectrum-scale-csi-attacher-5885c459cb-xcl6l      1/1     Running   0          39s
ibm-spectrum-scale-csi-operator-7f9657cfd5-pw8cb      1/1     Running   0          69s
ibm-spectrum-scale-csi-provisioner-6b69d67cff-79v2b   1/1     Running   0          39s
ibm-spectrum-scale-csi-resizer-6cc9f547d-t4zm7        1/1     Running   0          39s
ibm-spectrum-scale-csi-snapshotter-6cf9b7d69b-jdb6g   1/1     Running   0          39s
ibm-spectrum-scale-csi-tgxgx                          3/3     Running   0          39s
[root@saurabh-csi-master 0203]# oc get cso
NAME                     VERSION   SUCCESS
ibm-spectrum-scale-csi   2.9.0     True

Images: 
quay.io/ibm-spectrum-scale-dev/ibm-spectrum-scale-csi-operator:v2.9.0-020323
quay.io/ibm-spectrum-scale-dev/ibm-spectrum-scale-csi-driver:v2.9.0-020323
```

2. Create PVC
```
[root@saurabh-csi-master Independent]# cat sc.yaml
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
   name: ibm-spectrum-scale-csi-fileset-independent
provisioner: spectrumscale.csi.ibm.com
parameters:
    volBackendFs: "fs1"
    inodeLimit: "1024"
reclaimPolicy: Delete
[root@saurabh-csi-master Independent]# oc apply -f sc.yaml
storageclass.storage.k8s.io/ibm-spectrum-scale-csi-fileset-independent created
[root@saurabh-csi-master Independent]# oc apply -f pvc_independent_1.yaml ^C
[root@saurabh-csi-master Independent]# cat pvc_independent_1.yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: scale-fset-independent-pvc-2
spec:
  accessModes:
  - ReadWriteMany
  resources:
    requests:
      storage: 1Gi
  storageClassName: ibm-spectrum-scale-csi-fileset-independent
[root@saurabh-csi-master Independent]# oc apply -f pvc_independent_1.yaml
persistentvolumeclaim/scale-fset-independent-pvc-2 created
[root@saurabh-csi-master Independent]# oc get pvc -A
NAMESPACE                       NAME                           STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS                                 AGE
ibm-spectrum-scale-csi-driver   scale-fset-independent-pvc-2   Bound    pvc-ab5a44fd-7ebb-4aec-8942-a187597d4491   1Gi        RWX            ibm-spectrum-scale-csi-fileset-independent   19s
```

3. Unlink the fileset of the pvc
```
[root@saurabh-csi-scalegui fs1]# PATH=$PATH:/usr/lpp/mmfs/bin/
[root@saurabh-csi-scalegui fs1]# mmunlinkfileset fs1 pvc-ab5a44fd-7ebb-4aec-8942-a187597d4491 -f
Fileset pvc-ab5a44fd-7ebb-4aec-8942-a187597d4491 unlinked.
```

4. Delete the fileset of the pvc 
```
[root@saurabh-csi-scalegui fs1]# mmdelfileset fs1 pvc-ab5a44fd-7ebb-4aec-8942-a187597d4491 -f
Checking fileset ...
Checking fileset complete.
Deleting user files ...
 100.00 % complete on Fri Mar  3 01:32:16 2023  (      1024 inodes with total          4 MB data processed)
Deleting fileset ...
Fileset pvc-ab5a44fd-7ebb-4aec-8942-a187597d4491 deleted.
[root@saurabh-csi-scalegui fs1]# pwd
/ibm/fs1
```

5. Now delete the pvc 
```
[root@saurabh-csi-master Independent]# oc delete pvc scale-fset-independent-pvc-2
persistentvolumeclaim "scale-fset-independent-pvc-2" deleted
```

6. Check for the symlink path , it still exist  
```
[root@saurabh-csi-scalegui .volumes]# pwd
/ibm/fs1/spectrum-scale-csi-volume-store/.volumes
[root@saurabh-csi-scalegui .volumes]# ls -ltra | grep pvc-ab5a44fd-7ebb-4aec-8942-a187597d4491
lrwxrwxrwx 1 root root    95 Mar  3 01:29 pvc-ab5a44fd-7ebb-4aec-8942-a187597d4491 -> /ibm/fs1/pvc-ab5a44fd-7ebb-4aec-8942-a187597d4491/pvc-ab5a44fd-7ebb-4aec-8942-a187597d4491-data
```

## Expected behavior
Symlink path should be deleted if fileset is already deleted

**Comment by amdabhad** [comment]
Hi @deeghuge, this is not fixed yet.

**Comment by saurabhwani5** [comment]
closing as issue is fixed in #1095
